clients/src/main/java/org/apache/kafka/common/acl/AccessControlEntry.java

```diff
@@ -17,37 +17,61 @@
 
 package org.apache.kafka.common.acl;
 
+import org.apache.kafka.common.annotation.InterfaceStability;
+
 import java.util.Objects;
 
 /**
- * Represents an access control entry.  ACEs are a tuple of principal, host,
- * operation, and permissionType.
+ * Represents an access control entry.  ACEs are a tuple of principal, host, operation, and permissionType.
+ *
+ * The API for this class is still evolving and we may break compatibility in minor releases, if necessary.
  */
+@InterfaceStability.Evolving
 public class AccessControlEntry {
     final AccessControlEntryData data;
 
+    /**
+     * Create an instance of an access control entry with the provided parameters.
+     *
+     * @param principal non-null principal
+     * @param host non-null host
+     * @param operation non-null operation, ANY is not an allowed operation
+     * @param permissionType non-null permission type, ANY is not an allowed type
+     */
     public AccessControlEntry(String principal, String host, AclOperation operation, AclPermissionType permissionType) {
         Objects.requireNonNull(principal);
         Objects.requireNonNull(host);
         Objects.requireNonNull(operation);
-        assert operation != AclOperation.ANY;
+        assert(operation != AclOperation.ANY);
         Objects.requireNonNull(permissionType);
         assert permissionType != AclPermissionType.ANY;
```

Comment:
Should change this one as well if you are going to change the assert above to have parentheses